### ISSUE TITLE
D72R1 mcds engine not modelling multiple covariates

### ISSUE THREAD

**Issue by rjcamp** [DistanceDevelopment/distance-bugs#183]
Running single covariates the mcds engine works correctly, but get the error ** Warning: Invalid or missing covariate. ** when running two or more covariates in a model.

[dst64B7.txt](https://github.com/DistanceDevelopment/distance-bugs/files/2361524/dst64B7.txt)
[dst64E8.txt](https://github.com/DistanceDevelopment/distance-bugs/files/2361525/dst64E8.txt)

**Comment by LHMarshall** [comment]
Investigated the Distance projects and concluded that this is most likely a data issue rather than a software issue. 

The project which gave an error in 7.2 also gave an error in 7.1. A project which ran successfully in 7.1 with 2 factor level covariates was tested and still ran successfully in 7.2.

**Comment by LHMarshall** [comment]
Actually... have found something pretty weird. Which factor covariate goes first appears to affect the success... these are the same analyses I just switched the order of othe two covariates. In Distance one factor is stored as text the other is stored as integer. Data files show that only difference is that the two covariates have been switched in order.

**Comment by dill** [comment]
Some of the sensitivity is due to some of the covariate classification not quite being right, @rjcamp is following-up on this and we'll see how much this is an issue...

**Comment by dill** [comment]
Also unclear here if the issue is that `ObsYr` only has one level in the data (in my understanding of the MCDS command file at least, see below), yet is being included as a covariate but I don't know if that would raise the error Rick reports.

from the command file attached above:
```
Factor /Name=ObsYr /Levels=1 /Labels=1987;
```

then later

```
Estimator /Key=HA /Adjust=CO /NAP=0 /Covariates=Obs, ObsYr;
```

**Comment by lenthomas** [comment]
I don't think this is the problem per se as a simple example (below) with just one factor covar with one level runs okay with the warning 

`** Warning: There is only one level for factor covariate COV. A minimum of two levels is required for estimation; hence this covariate will be omitted. **
`
[dstB47.txt](https://github.com/DistanceDevelopment/distance-bugs/files/5851287/dstB47.txt)
[dstB68.txt](https://github.com/DistanceDevelopment/distance-bugs/files/5851288/dstB68.txt)
